File: docs/breez-sdk/snippets/wasm/lnurl_withdraw.ts
import { type Wallet, parseInput } from '@breeztech/breez-sdk-spark'

const exampleLnurlWithdraw = async (wallet: Wallet) => {
  // ANCHOR: lnurl-withdraw
  // Endpoint can also be of the form:
  // lnurlw://domain.com/lnurl-withdraw?key=val
  const lnurlWithdrawUrl =
    'lnurl1dp68gurn8ghj7mr0vdskc6r0wd6z7mrww4exctthd96xserjv9mn7um9wdekjmmw843xxwpexdnxzen9vgunsvfexq6rvdecx93rgdmyxcuxverrvcursenpxvukzv3c8qunsdecx33nzwpnvg6ryc3hv93nzvecxgcxgwp3h33lxk'

  const input = await parseInput(lnurlWithdrawUrl)
  if (input.type === 'lnurlWithdraw') {
    // Amount to withdraw in sats between min/max withdrawable amounts
    const amountSats = 5_000
    const withdrawRequest = input
    const optionalCompletionTimeoutSecs = 30

    const response = await wallet.lnurl.withdraw({
      amountSats,
      withdrawRequest,
      completionTimeoutSecs: optionalCompletionTimeoutSecs
    })

    const payment = response.payment
    console.log(`Payment: ${JSON.stringify(payment)}`)
  }
  // ANCHOR_END: lnurl-withdraw
}

File: docs/breez-sdk/snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/LnurlWithdraw.kt
package com.example.kotlinmpplib

import breez_sdk_spark.*
class LnurlWithdraw {
    suspend fun lnurlWithdraw(sdk: BreezSdk) {
        // ANCHOR: lnurl-withdraw
        // Endpoint can also be of the form:
        // lnurlw://domain.com/lnurl-withdraw?key=val
        val lnurlWithdrawUrl = "lnurl1dp68gurn8ghj7mr0vdskc6r0wd6z7..."
        try {
            val inputType = sdk.parse(lnurlWithdrawUrl)
            if (inputType is InputType.LnurlWithdraw) {
                // Amount to withdraw in sats between min/max withdrawable amounts
                val amountSats = 5_000.toULong()
                val withdrawRequest = inputType.v1
                val optionalCompletionTimeoutSecs = 30.toUInt()

                val request = LnurlWithdrawRequest(
                    amountSats,
                    withdrawRequest,
                    optionalCompletionTimeoutSecs
                )
                val response = sdk.lnurlWithdraw(request)

                val payment = response.payment
                // Log.v("Breez", "Payment: $payment")
            }
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: lnurl-withdraw
    }
}